src/deluge/util/functions.cpp

```diff
@@ -2690,28 +2690,32 @@ bool shouldAbortLoading() {
 
 // Must supply a char[5] buffer. Or char[30] for OLED.
 void getNoteLengthNameFromMagnitude(char* text, int32_t magnitude, bool clarifyPerColumn) {
+	char noteLengthName[64];
+	StringBuf noteLengthBuf(noteLengthName, 64);
```

Comment:
I think this can allocate

Reply:
StringBuf's entire point of existence is to provide a nice interface for high level interaction with a manually managed string. 

The entire implementation is 20 lines. where would it be allocating?

```
class StringBuf {
	// Not templated to optimize binary size.
public:
	StringBuf(char* buf, size_t capacity) : capacity_(capacity), buf_(buf) {}

	void append(const char* str) { ::strncat(buf_, str, capacity_ - size() - 1); }
	void append(char c) { ::strncat(buf_, &c, 1); }
	void clear() { buf_[0] = 0; }

	// TODO: Validate buffer size. This will overflow
	void appendInt(int i, int minChars = 1) { intToString(i, buf_ + size(), minChars); }
	void appendHex(int i, int minChars = 1) { intToHex(i, buf_ + size(), minChars); }

	[[nodiscard]] char* data() { return buf_; }
	[[nodiscard]] const char* data() const { return buf_; }
	[[nodiscard]] const char* c_str() const { return buf_; }

	[[nodiscard]] std::size_t capacity() const { return capacity_; }
	[[nodiscard]] std::size_t size() const { return ::strlen(buf_); }

	[[nodiscard]] bool empty() const { return buf_[0] == 0; }

	bool operator==(const char* rhs) const { return strcmp(buf_, rhs) == 0; }
	bool operator==(StringBuf const& rhs) const { return strcmp(buf_, rhs.c_str()) == 0; }

private:
	size_t capacity_;
	char* buf_;
};
```

Reply:
Oh my bad, thought it was the std string buf

Reply:
Could you add a comment mentioning that it's totally different?

Reply:
Disregard, I'm gonna just rename that class later instead